File: array/add-two-numbers/enrique/add-two-numbers.go
package enrique

func twoSum(nums []int, target int) []int {
	var indexA, indexB int

	valueToIdx := map[int]int{}
	for i, v := range nums {
		difference := target - v

		if idx, ok := valueToIdx[difference]; ok {
			indexA = i
			indexB = idx

			break
		}

		valueToIdx[v] = i
	}

	return []int{indexA, indexB}
}

File: array/add-two-numbers/enrique/add-two-numbers.py
from collections import namedtuple

IdxMetadata = namedtuple('IdxMetadata', ['indices', 'value_needed'])

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nums_map = {}
        for idx, value in enumerate(nums):
            complement = target - value
            if complement in nums_map:
                return [nums_map[complement], idx]

            nums_map[value] = idx
